### PR TITLE
Allow update of proxmox container configuration

### LINKED ISSUE #7145: Proxmox module does not have an option to update the config
### Summary

In the proxmox_kvm module there is an update flag to allow updating the config of an existing VM, the proxmox module does not have an equivalent option. So once a container is created you can't modify it.

### Issue Type

Feature Idea

### Component Name

proxmox

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by etrombly**
##### SUMMARY

Add an update parameter to the proxmox module allowing you to update the configuration of an already created container. This was already possible for VMs in the proxmox_kvm module.

Fixes #7145 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

proxmox

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by etrombly** [comment]
Just realized I had VSCode set to auto-format. Sorry about that. If it makes the changes too hard to review I can revert the code formatting.

**Comment by felixfontein** [comment]
@etrombly thanks for your contribution! Please revert the formatting changes. Thanks!

**Comment by etrombly** [comment]
formatting changes have been reverted

File: plugins/modules/proxmox.py [hunk 1]
```diff
@@ -733,8 +798,43 @@ def main():
...
+                                              searchdomain=module.params["searchdomain"],
+                                              features=",".join(module.params["features"])
+                                              if module.params["features"] is not None
+                                              else None,
+                                              description=module.params["description"],
+                                              hookscript=module.params["hookscript"],
+                                              timezone=module.params["timezone"],
+                                              tags=module.params["tags"])
+                        module.exit_json(
+                            changed=True,
```
Comment (felixfontein) [inline plugins/modules/proxmox.py:822]: Is it possible to detect whether there are no changes? If yes, that should be done. If not, the documentation should be very clear that the module is not able to detect this.

**Review by felixfontein — COMMENTED** [review]
A general comment from my side. I hope that someone with more Proxmox knowledge also takes a look at this PR :)

File: plugins/modules/proxmox.py (hunk 1 above)
Comment (etrombly) [reply plugins/modules/proxmox.py:822]: Good catch, I can try fetching the current config from the API and diffing against the new requested config.

Comment (felixfontein) [reply plugins/modules/proxmox.py:822]: BTW, the proxmox_kvm module also has an `update` parameter, and it seems to have the same problem - see #1965. https://github.com/ansible-collections/community.general/issues/1965#issuecomment-1043439573 might be relevant here as well.

Comment (etrombly) [reply plugins/modules/proxmox.py:849]: I'm comparing against the current running config now. I tested against the majority of config options and everything seemed good, but didn't test every option.

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this at the end of this week (i.e. around upcoming Friday).

**Comment by felixfontein** [comment]
@etrombly thanks for your contribution!

**Pull request merged**